### DISCUSSION TITLE
A way to insert text with (insert mode) keymapping

### DISCUSSION THREAD

**Discussion by tmke8** [helix-editor/helix#10212] [General]
For markdown files, I like to have a keybinding where when I type `` `p`` it's expanded to `` ```python`` (similarly, `` `r`` to `` ```rust`` and so on).

Currently, I do this with
```toml
[keys.insert]
"`" = { p = ["normal_mode", ":pipe echo '```python'", "collapse_selection", "insert_mode"] }
```
This does work, but it seems a bit convoluted.

As far as I know there is currently no other way. (Correct me if I'm wrong.)

I would imagine a solution like this:
```toml
[keys.insert]
"`" = { p = ":insert_text ```python" }
```

EDIT:
The current work-around is
```toml
[keys.insert]
"`" = { p = [":insert-output echo -n '```python'"] }
```

**Comment by the-mikedavis** [comment]
You could simplify that keymap with `:insert-output`/`:append-output` https://github.com/helix-editor/helix/pull/2589

It's probably best to solve this with #1383 in the long run.

**Comment by tmke8** [comment]
Ah, I was using release 22.05 so I didn't have those commands. Sorry!

**Comment by tmke8** [comment]
On closer inspection, those commands are also based on calling shell commands. So, it's basically the same issue with them.

**Comment by sudormrfbin** [comment]
I would say snippet support (#1178) is the best way to solve this since you want some text to expand to an arbitrary longer text.
